components/resources/quizzes/shared/SkillSelections.tsx

```diff
@@ -13,20 +18,49 @@ type SkillSelectionsProps = {
 
 const SkillSelections: React.FC<SkillSelectionsProps> = ({
   selections,
+  selected,
+  setSelected,
+  setQuizResponses,
+  currentStage,
   onNextClick,
   onBackClick,
   progress,
   title,
   body,
 }) => {
-  const [selected, setSelected] = useState<string[]>([]);
+  const currentSelections = selected[currentStage] || [];
+
   const handleSelection = (selection: string) => {
-    if (selected.includes(selection)) {
-      setSelected(selected.filter((s) => s !== selection));
+    const selectionIndex = currentSelections.indexOf(selection);
```

Comment:
what does indexOf do that map or filter can't? Asking because ive not used it and am curious.

Reply:
I was more interested in index #. Hence - I went with this approach.